g2p/transducer/__init__.py

```diff
@@ -692,10 +858,10 @@ def apply_rules(self, to_convert: str):  # noqa: C901
                 except IndexError:
                     following = None
                 if previous:
-                    edge[1] = previous[-1][1]
+                    tg.edges[i] = (edge[0], previous[-1][1])
                 elif following:
-                    edge[1] = following[0][1]
-        tg.edges = list(dict.fromkeys([tuple(x) for x in tg.edges]))
+                    tg.edges[i] = (edge[0], following[0][1])
+        tg.edges = list(OrderedDict.fromkeys((i, j) for i, j in tg.edges))
```

Comment:
The reason is that otherwise it isn't clear that we want to preserve the order, and someone might be tempted to replace the `dict` with a `set` (which is what I did, and then realized that this doesn't work)